retail-managment-system/rms-ui/src/app/domain/user/user.repository.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { ResourceService } from 'src/app/core/services/resource.service';
+import { User } from './models/user';
+import { HttpClient } from '@angular/common/http';
+import { catchError, Observable } from 'rxjs';
+import { Role } from '../role/models/role';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserRepository extends ResourceService<User>{
+
+  constructor(httpClient: HttpClient) {
+    super(httpClient)
+   }
+
+   getResourceUrl(): string {
+     return 'user'
+   }
+
+   assignRoleToServerModel(entity:Role):any{
+    return {
+      id: entity.role.id
+    }
+   }
+   assignRole(resource:Role,id:number):Observable<any>{
+      return this.httpClient
+        .post(`${this.APIUrl}/${id}/role`, this.assignRoleToServerModel(resource))
+        .pipe(
+          catchError((err) => {
+            throw new Error(err.message);
+          })
+        );
+    }
+
+    deleteRole(id: string | number): Observable<any> {
```

Comment:
use meaningful name (specific name) deleteRoleOfUser
because it isn't delete role but delete role of use